libc/src/time/strftime.cpp

```diff
@@ -0,0 +1,25 @@
+//===-- Implementation of strftime function -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/time/strftime.h"
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+#include "src/time/time_utils.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+using LIBC_NAMESPACE::time_utils::TimeConstants;
+
+LLVM_LIBC_FUNCTION(size_t, strftime,
+                   (char *__restrict, size_t, const char *__restrict,
+                    const struct tm *)) {
+  // TODO: Implement this for the default locale.
```

Comment:
I think we should make these fail more assertively than just returning zero. I'd say setting `libc_errno` to `ENOSYS` and returning `-1` would work.

Reply:
ENOSYS isn't a generic errno value, so probably can't use it for all implementations.

Reply:
Given that this is a posix function, it should be safe to use `ENOSYS` since that's a posix error.